python/lsst/analysis/tools/tasks/catalogMatch.py

```diff
@@ -0,0 +1,408 @@
+        dtype=float,
+        default=1.0,
+    )
+
+
+class AstropyMatchTask(pipeBase.Task):
+    """ A task for running the astropy matcher `match_to_catalog_sky` on
+    between target and reference catalogs."""
+
+    ConfigClass = AstropyMatchConfig
+
+    def run(self, refCatalog, targetCatalog):
+        """Run matcher
+
+        Parameters
+        ----------
+        refCatalog: `pandas.core.frame.DataFrame`
+            The reference catalog with coordinates in degrees
+        targetCatalog: `pandas.core.frame.DataFrame`
+            The target catalog with coordinates in degrees
+
+        Returns
+        -------
+        `pipeBase.Struct` containing:
+            refMatchIndices: `numpy.ndarray`
+                Array of indices of matched reference catalog objects
+            targetMatchIndices: `numpy.ndarray`
+                Array of indices of matched target catalog objects
+            separations: `astropy.coordinates.angles.Angle`
+                Array of angle separations between matched objects
+        """
+        refCat_ap = SkyCoord(ra=refCatalog['coord_ra'] * u.degree,
+                             dec=refCatalog['coord_dec'] * u.degree)
+
+        sourceCat_ap = SkyCoord(ra=targetCatalog['coord_ra'] * u.degree,
+                                dec=targetCatalog['coord_dec'] * u.degree)
+
+        id, d2d, d3d = refCat_ap.match_to_catalog_sky(sourceCat_ap)
+
+        goodMatches = d2d.arcsecond < self.config.maxDistance
+
+        refMatchIndices = np.flatnonzero(goodMatches)
+        targetMatchIndices = id[goodMatches]
+
+        separations = d2d[goodMatches].arcsec
+
+        return pipeBase.Struct(refMatchIndices=refMatchIndices,
+                               targetMatchIndices=targetMatchIndices,
+                               separations=separations)
+
+
+class CatalogMatchConnections(pipeBase.PipelineTaskConnections, dimensions=("tract", "skymap"),
+                              defaultTemplates={"targetCatalog": "objectTable_tract",
+                                                "refCatalog": "astrometryRefCat"}):
+
+    catalog = pipeBase.connectionTypes.Input(
+        doc="The tract-wide catalog to make plots from.",
+        storageClass="DataFrame",
+        name="{targetCatalog}",
+        dimensions=("tract", "skymap"),
+        deferLoad=True
+    )
+
+    refCat = pipeBase.connectionTypes.PrerequisiteInput(
+        doc="The reference catalog to match to loaded input catalog sources.",
+        name="gaia_dr2_20200414",
+        storageClass="SimpleCatalog",
+        dimensions=("skypix",),
+        deferLoad=True,
+        multiple=True
+    )
+
+    skymap = pipeBase.connectionTypes.Input(
+        doc="The skymap for the tract",
+        storageClass="SkyMap",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        dimensions=("skymap",)
+    )
+
+    matchCatalog = pipeBase.connectionTypes.Output(
+        doc="Catalog with matched target and reference objects with separations",
```

Comment:
I think I would prefer matchedCatalog as it reinforces that this is the one that has been matched rather than the one to match but I don't think it really matters.